include/picongpu/fields/currentInterpolation/Binomial/Binomial.hpp

```diff
@@ -51,35 +62,140 @@ namespace currentInterpolation
             T_DataBoxJ const fieldJ
         )
         {
-            DataSpace< dim > const self;
             using TypeJ = typename T_DataBoxJ::ValueType;
+            using DS = DataSpace< dim >;
+
+            // weighting for original value, i.e. center element of a cell
+            constexpr float_X M = 8.0;
+            // weighting for nearest neighbours, i.e. cells sharing a face with the center cell
+            constexpr float_X S = 4.0;
+            // weighting for next to nearest neighbours, i.e. cells sharing an edge with the center cell
+            constexpr float_X D = 2.0;
+            // weighting for farthest neighbours, i.e. cells sharing a corner with the center cell
+            constexpr float_X T = 1.0;
+
+            TypeJ averagedJ =
+                // sum far neighbours, i.e. corner elements, weighting T
+                T * (
+                    fieldJ( DS( -1, -1, -1 ) ) + fieldJ( DS( +1, -1, -1 ) ) + fieldJ( DS( -1, +1, -1 ) ) + fieldJ( DS( +1, +1, -1 ) ) +
+                    fieldJ( DS( -1, -1, +1 ) ) + fieldJ( DS( +1, -1, +1 ) ) + fieldJ( DS( -1, +1, +1 ) ) + fieldJ( DS( +1, +1, +1 ) )
+                ) +
+                // sum next to nearest neighbours, i.e. edge elements, weighting D
+                D * (
+                    fieldJ( DS( -1, -1, 0 ) ) + fieldJ( DS( +1, -1, 0 ) ) + fieldJ( DS( -1, +1, 0 ) ) + fieldJ( DS( +1, +1, 0 ) ) +
+                    fieldJ( DS( -1, 0, -1 ) ) + fieldJ( DS( +1, 0, -1 ) ) + fieldJ( DS( -1, 0, +1 ) ) + fieldJ( DS( +1, 0, +1 ) ) +
+                    fieldJ( DS( 0, -1, -1 ) ) + fieldJ( DS( 0, +1, -1 ) ) + fieldJ( DS( 0, -1, +1 ) ) + fieldJ( DS( 0, +1, +1 ) )
+                ) +
+                // sum next neighbours, i.e. face elements, weighting S
+                S * (
+                    fieldJ( DS( -1, 0, 0 ) ) + fieldJ( DS( +1, 0, 0 ) ) +
+                    fieldJ( DS( 0, -1, 0 ) ) + fieldJ( DS( 0, +1, 0 ) ) +
+                    fieldJ( DS( 0, 0, -1 ) ) + fieldJ( DS( 0, 0, +1 ) )
+                ) +
+                // add original value, i.e. center element, weighting M
+                M * (
+                    fieldJ( DS( 0, 0, 0 ) )
+                );
+
+            /* calc average by normalizing weighted sum In 3D there are:
+             *   - original value with weighting M
+             *   - 6 nearest neighbours with weighting S
+             *   - 12 next to nearest neighbours with weighting D
+             *   - 8 farthest neighbours with weighting T
+             */
+            averagedJ /= M + 6.*S + 12.*D + 8.*T;
```

Comment:
not sure if this is requited, since the compiles knows that all is const at compile time and can transform this expression easily. I will check the PTX code.

Reply:
@sbastrakov @ax3l 
I checked how the compiler is handling this code. In the case we compile with fast math nvcc uses `div.approx.f32` (fast math division) and if we compile without fast math it is transformed to a multiplication `mul.f32`. In both cases, the compiler fully evaluate the rhs into a constant value.
So there is no need to explicitly put it into a const expression.

Reply:
@psychocoderHPC sorry for a potentially stupid question, but isn't div.approx.f32 actually a reciprocal + multiplication?

Reply:
Yes, the ISA guide said the following:

```
div.approx.f32 implements a fast approximation to divide, computed as d = a
* (1/b) . For b in [2 -126 , 2 126 ], the maximum ulp error is 2.
```

> Then it kinda does not make sense that fast math results in more operations.

What do you mean with that?

Reply:
I'm probably missing smth trivial, but to me it looks like this: with fast math enabled the compiler uses div.approx.f32, which is one multiplication + one reciprocal, and to me it looks as if the reciprocal of this constant is performed in run time. While without fast math, it is just a multiplication, which looks simply faster than in the "fast math" mode.

Reply:
`div.approx.f32` is performed in one operation and the throughput should be equal to the normal multiplication (if I remember correctly). IMO the compiler knows best how to transform this operation. If we explicitly write it as multiplication we reduce the possibility for optimizations e.g. on x86 architectures.

Reply:
the idea is to make the operation a pure multiplication, which will always be better than a division on any arch.

Reply:
In x86 architecture you have more than 64bit for 64bit floating points if this value is stored.
Since `x/y` is not equal to `x * 1/y` the explicit multiplication can increase the precision error. 
Therefore the compiler can deside based on the compiler parameters given by the user and the hardware if the optimization should be used or not.

Reply:
That's a different and valid argument about floating-point precision at cutoff and not the optimization of "number of operations" argument discussed above.

Regarding floating-point precision, is luckily not significant in this factor where we are already smoothing out out data to violate charge conservation significantly.